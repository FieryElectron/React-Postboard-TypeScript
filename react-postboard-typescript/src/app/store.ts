import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import signInPageReducer from '../components/SignInPage/signInPageSlice';
import postBoardPageReducer from '../components/PostBoardPage/postBoardPageSlice';

export const store = configureStore({
  reducer: {
    signInPage: signInPageReducer,
    postBoardPage: postBoardPageReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
