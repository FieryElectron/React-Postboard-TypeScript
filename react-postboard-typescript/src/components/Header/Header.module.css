.headerContainer {
    background-color: #FAFAFA;
    border-bottom: 2px solid gray;
}

.header {
    position: relative;
    /* left: 40%; */
    margin: 0;
    padding: 8px;
}

.profile{
    position: absolute;

    right: 5px;
    top:10px;

    display: flex;
    justify-content: center;
    align-items: center;
}