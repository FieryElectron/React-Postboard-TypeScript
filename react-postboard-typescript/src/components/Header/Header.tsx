import styles from './Header.module.css';

import { FaUserAlt } from 'react-icons/fa';
import { useLocation } from 'react-router-dom';
import Button from '../Button/Button'
import { useNavigate } from 'react-router-dom';
import axios from 'axios';
import { useAppSelector } from '../../app/hooks';

import {
    selectLogout,
} from '../SignInPage/signInPageSlice';

import {
    selectUsername,
} from '../PostBoardPage/postBoardPageSlice';

type Props = {
    title:string
}

const Header = ({ title } : Props) => {
    const username = useAppSelector(selectUsername);
    const logoutUrl = useAppSelector(selectLogout);

    const location = useLocation();
    const navigate = useNavigate();


    const logoutOnClick = async () => {
        await axios.delete(logoutUrl, {withCredentials: true});
        navigate('/');
    }

    return (
        <div data-testid='headerContainer' className={styles.headerContainer}>
            <h1 data-testid='header' className={styles.header}>{title}</h1>
            {location.pathname === '/postboard' && username !== '' && <span className={styles.profile}><FaUserAlt color='blue' size={30} /> {username}
            <Button color='red' text='Log out' onClick={logoutOnClick}/>
            </span>}
        </div>
    )
}

Header.defaultProps = {
    title: 'title',
}

export default Header