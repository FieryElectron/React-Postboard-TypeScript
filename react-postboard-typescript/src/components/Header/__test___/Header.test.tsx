import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import Header from '../Header';
import { Provider } from 'react-redux';

import { BrowserRouter as Router} from 'react-router-dom'
import { store } from '../../../app/store';

let container: HTMLElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container.remove();
});

it('empty test', ()=>{

});

it('renders Header without crashing', () => {
  act(() => {
    render(<Provider store={store}><Router><Header/></Router></Provider>);
  });
});

it('renders Header class correctly', () => {
  act(() => {
    render(<Provider store={store}><Router><Header/></Router></Provider>);
  });

  const headerContainer = document.querySelector('[data-testid=headerContainer]')!;
  const header = document.querySelector('[data-testid=header]')!;

  expect(headerContainer).toHaveAttribute('class', 'headerContainer');
  expect(header).toHaveAttribute('class', 'header');
});

it('renders Header title correctly', () => {
  act(() => {
    render(<Provider store={store}><Router><Header title='PostBoard'/></Router></Provider>);
  });

  const header = document.querySelector('[data-testid=header]')!;

  expect(header).toHaveTextContent('PostBoard');
});

it("matchs snapshot 1", ()=>{
    const tree = renderer.create(<Provider store={store}><Router><Header title='Sign In'/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
}); 
 
it("matchs snapshot 2", ()=>{
    const tree = renderer.create(<Provider store={store}><Router><Header title='PostBoard'/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
}); 


