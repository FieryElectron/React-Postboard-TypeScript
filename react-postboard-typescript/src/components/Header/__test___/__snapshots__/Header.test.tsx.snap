// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div
  className="headerContainer"
  data-testid="headerContainer"
>
  <h1
    className="header"
    data-testid="header"
  >
    Sign In
  </h1>
</div>
`;

exports[`matchs snapshot 2 1`] = `
<div
  className="headerContainer"
  data-testid="headerContainer"
>
  <h1
    className="header"
    data-testid="header"
  >
    PostBoard
  </h1>
</div>
`;
